satpy/readers/clavrx.py

```diff
@@ -305,16 +237,257 @@ def _read_axi_fixed_grid(self, l1b_attr):
+        self.sensor = self.nc.attrs.get('sensor', None)
+
+    def _get_ds_info_for_data_arr(self, var_name):
+        ds_info = {
+            'file_type': self.filetype_info['file_type'],
+            'name': var_name,
+            'coordinates': ["longitude", "latitude"]
+        }
+        return ds_info
+
+    def _is_2d_yx_data_array(self, data_arr):
+        has_y_dim = data_arr.dims[0] == "y"
+        has_x_dim = data_arr.dims[1] == "x"
+        return has_y_dim and has_x_dim
+
+    def _available_new_datasets(self, handled_vars):
+        """Metadata for available variables other than BT."""
+        possible_vars = list(self.nc.items()) + list(self.nc.coords.items())
+        for var_name, data_arr in possible_vars:
+            if var_name in handled_vars:
+                continue
+            if data_arr.ndim != 2:
+                # we don't currently handle non-2D variables
+                continue
+            if not self._is_2d_yx_data_array(data_arr):
+                # we need 'traditional' y/x dimensions currently
+                continue
+
+            ds_info = self._get_ds_info_for_data_arr(var_name)
+            yield True, ds_info
+
+    def available_datasets(self, configured_datasets=None):
+        """Dynamically discover what variables can be loaded from this file.
+
+        See :meth:`satpy.readers.file_handlers.BaseHandler.available_datasets`
+        for more information.
+
+        """
+        handled_vars = set()
+        for is_avail, ds_info in (configured_datasets or []):
+            if is_avail is not None:
+                # some other file handler said it has this dataset
+                # we don't know any more information than the previous
+                # file handler so let's yield early
+                yield is_avail, ds_info
+                continue
+            if self.file_type_matches(ds_info['file_type']):
+                handled_vars.add(ds_info['name'])
+            yield self.file_type_matches(ds_info['file_type']), ds_info
+        yield from self._available_new_datasets(handled_vars)
+
+    def _is_polar(self):
+        l1b_att, inst_att = (str(self.nc.attrs.get('L1B', None)),
+                             str(self.nc.attrs.get('sensor', None)))
+
+        return (inst_att != 'AHI' and 'GOES' not in inst_att) or (l1b_att is None)
+
+    def get_area_def(self, key):
+        """Get the area definition of the data at hand."""
+        if self._is_polar():  # then it doesn't have a fixed grid
+            return super(CLAVRXNetCDFFileHandler, self).get_area_def(key)
+
+        l1b_att = str(self.nc.attrs.get('L1B', None))
+        return _CLAVRxHelper._read_axi_fixed_grid(self.filename, l1b_att)
+
+    def get_metadata(self, attrs, ds_info):
+        """Get metadata."""
+        i = {}
+        i.update(attrs)
+        i.update(ds_info)
+
+        flag_meanings = i.get('flag_meanings', None)
+        if not i.get('SCALED', 1) and not flag_meanings:
```

Comment:
The NetCDF files use `SCALED`? Not a more CF standard `scale_factor`? :cry:

Reply:
I would assume that as clavrx development continues that things become more and more CF-y so if `scale_factor` and `add_offset` are there, let's use those.

Reply:
Actually, I am not sure I should change this.  This isolates the variables that are flags well without pulling in extra variables which are not.  If I would switch to scale_factor, there is one an extra variable which would be picked up that probably shouldn't be assigned a flag value unknown attribute.

Reply:
I think I'm misunderstanding what the original purpose of this code is. What do you mean by "extra variables"? Aren't SCALED and scale_factor just attributes in the dictionary?

I realize now that this if statement isn't controlling if scale/offset are applied which is what I originally thought it was. I'm now fine with however you have to make this work. I assumed SCALED and scale_factor were the same thing and that you were then going to use SCALED to scale the data later. So...maybe just ignore that I said anything?

Reply:
I know this is no excuse, but that was in the code before I started working on it.  As far as I can see, this does not do anything to the data scaling, that is handled with scale_factor and add_offset in get_dataset.  This adds some attribute information to the data variable: if these are flags, and for some reason, they do not have "flag_meaning" metadata to describe them, add a "meaning unknown" attribute to the metdata.  All the flag variables + 1 "duplicate" variable have a scale_factor of 1.0, but the duplicate variable is SCALED as 1b versus the flags which are 0b.

Reply:
Sounds good.